lib/greet.js

```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = exports = {};
+var name = process.argv[2];
+
+exports.sayHello = function(name) {
+  if(arguments.length === 0) throw new Error('name not provided');
+  console.log(`hello ${name}!`);
+  return `hello ${name}!`;
+};
+
+exports.sayHello(name);
```

Comment:
Since you are using module.exports you don't need to explicitly export the function as you did on line 12. Also, it's best to use module.exports = function ( ) when exporting a single method :) module.exports = exports = { } is used for exporting multiple methods. For instance, if you had another method attached to the exports object, then using module exports how you did on line 3 would be just fine.